### ISSUE TITLE
Issue with Lab 4 (implementing compareTo method)

### ISSUE THREAD

**Issue by wongjingwen** [nus-cs2030/2223-s2#156]
# Summary

The booking object has a toString method that returns the cheapest service offered by the driver. However, level 5 requires us to compare 2 different booking obj. How do we go about comparing the individual services of each cab with one another?

**Comment by choongzhanhong** [comment]
I'm not sure if it's the best way to do it, but I used Booking to encapsulate the best rate (based on which services the driver offers).

So e.g. NormalCab only offers 2 services, Booking can have a method to check through the driver's services with the data from Request and store the best fare as a property. 
Comparing to other Booking most likely uses a getter for that property.

If anyone has a better idea, can share!

**Comment by wongjingwen** [comment]
> So e.g. NormalCab only offers 2 services, Booking can have a method to check through the driver's services with the data from Request and store the best fare as a property.
This would be done in the constructor?

**Comment by choongzhanhong** [comment]
Yeah, I did it in the constructor then stored the best fare and service as properties in the Booking class

**Comment by wongjingwen** [comment]
The compareTo method only compares the best fare b/w both Booking but not the other services.

**Comment by fy17ohhh** [comment]
> The compareTo method only compares the best fare b/w both Booking but not the other services. How do we ensure that the other services are printed as well

same qn here, how can we compare Booking class with higher fare printed as well? when we construct booking class we are not including that info.

**Comment by Jahhlen** [comment]
Maybe try overloading the driver constructor which allows a method within the driver to return a new driver with just 1 service in the "serviceList". Then you can force booking to use that service in level5

**Comment by sharlynntan** [comment]
Alternatively, you can overload the constructor of Booking (with a different parameter length) to return a create a new booking of the alternative service. (Currently there is only 2 services per driver, so you can do as such)

**Comment by haqanh** [comment]
> > The compareTo method only compares the best fare b/w both Booking but not the other services. How do we ensure that the other services are printed as well
> 
> same qn here, how can we compare Booking class with higher fare printed as well? when we construct booking class we are not including that info.

I think what you can try is to store a list of all the possible Driver-Service combinations? then, your level5.jsh would access these list and compare them. (i.e. as others have pointed out, you need to define a new method/constructor as the behavior specified in Level 4 only look at the best booking)

Hope this helps!

**Comment by wongjingwen** [comment]
> > > The compareTo method only compares the best fare b/w both Booking but not the other services. How do we ensure that the other services are printed as well
> > 
> > 
> > same qn here, how can we compare Booking class with higher fare printed as well? when we construct booking class we are not including that info.
> 
> I think what you can try is to store a list of all the possible Driver-Service combinations? then, your level5.jsh would access these list and compare them. (i.e. as others have pointed out, you need to define a new method/constructor as the behavior specified in Level 4 only look at the best booking)
> 
> Hope this helps!

For the toString method in Booking, does it still return the Best fare or do we need to modify?
